src/main/java/com/pages/BasePage.java

```diff
@@ -0,0 +1,12 @@
+package com.pages;
+
+import io.appium.java_client.android.AndroidDriver;
+
+public class BasePage {
```

Comment:
protected WebElement findElementByLocator(By locator) {
        wait.until(ExpectedConditions.visibilityOfElementLocated(locator));
        return this.driver.findElement(locator);
    }

protected void waitForElementVisible(By locator) {
        wait.until(ExpectedConditions.visibilityOfElementLocated(locator));
    }
//
    protected void waitForElementClickable(By locator) {
        wait.until(ExpectedConditions.elementToBeClickable(locator));
    }